### PR TITLE
Add universal sentence encoder

### PR DESCRIPTION
**Pull request by jerpint**
Adds the universal sentence encoder as the first input to the LSTM

### CODE REVIEW

File: train.py [hunk 1]
```diff
@@ -25,14 +23,20 @@
 num_features = context_embeddings.shape[2]
 raw_question = questions[0][0]['question']
 
-words_in_context = tf.placeholder(tf.float32, [time_steps, batch_size, num_features])
-lstm = tf.keras.layers.LSTM(units=3)
```
Comment (gmarceaucaron) [inline train.py]: From the doc: Note that this cell is not optimized for performance on GPU. Please use tf.compat.v1.keras.layers.CuDNNLSTM for better performance on GPU.
 But I think tf.keras.layers.CuDNNLSTM (https://www.tensorflow.org/versions/r1.14/api_docs/python/tf/keras/layers/CuDNNLSTM) can be used.

File: train.py [hunk 2]
```diff
@@ -25,14 +23,20 @@
 num_features = context_embeddings.shape[2]
 raw_question = questions[0][0]['question']
 
-words_in_context = tf.placeholder(tf.float32, [time_steps, batch_size, num_features])
```
Comment (gmarceaucaron) [inline train.py]: Are you sure about the dimensions? According to the [Keras API of LSTM](https://www.tensorflow.org/versions/r1.14/api_docs/python/tf/keras/layers/LSTM), the default value of the `time_major` parameter is false and so, this layer takes inputs in batch major mode. Moreover, this is consistent with the `tf.losses.softmax_cross_entropy` function.

File: train.py (hunk 1 above)
Comment (jerpint) [reply train.py]: oh I did not know this, good catch!

File: train.py [hunk 3]
```diff
@@ -25,14 +23,20 @@
...
 
-question_embedding_placeholder = tf.placeholder(tf.float32, [batch_size, 512])
+question_embedding_use = embed_model([raw_question])
+dense_linear = tf.keras.layers.Dense(units=num_features)
+question_embedding = dense_linear(question_embedding_use)
+question_embedding = tf.reshape(question_embedding, [batch_size, 1, num_features])
+
+context_embedding = tf.placeholder(tf.float32, [time_steps, batch_size, num_features])
+
+embedded_question_context = tf.concat([question_embedding, context_embedding], axis=0)
```
Comment (gmarceaucaron) [inline train.py]: the dimensions of the concatenation operation are not consistent. On axis 0, `question_embedding` has dimension `batch_size` while `context_embedding` has dimension `time_steps`.

File: train.py (hunk 1 above)
Comment (jerpint) [reply train.py]: This will not work on CPU, so we will keep it as is for debugging and switch over when training on a GPU, maybe add a flag?

File: train.py (hunk 2 above)
Comment (jerpint) [reply train.py]: Good point, although here this line was removed. In the latest commit I just pushed, everything has been set to be `batch_size` major consistently, i.e. `[batch_size, time_steps, num_features]`

File: train.py (hunk 3 above)
Comment (jerpint) [reply train.py]: I have in the latest push changed the dimensions to be consistent with the ordering `[batch_size, time_steps, num_features]`
